src/daemon.py

```diff
@@ -231,7 +227,23 @@ def __process_task(self, task: AgentTask) -> None:
         :type task: AgentTask
         :raises RuntimeError: Task with unsupported type given.
         """
-        if task.type == TaskType.SEARCH:
+        if task.type == TaskType.RELOAD:
+            if (
+                self.plugin_config_version
+                == self.db.get_plugin_config_version()
+            ):
+                # This should never happen and suggests that version is not updated properly.
```

Comment:
IMO runtime error here is too harsh:
 - 1) configuration is only increasing, so this shouldn't cause a catastrophic failure and even if it's wrong, configuration will fix itself in the next version
 - 2) this kills the daemon

Maybe just logging.error() and continue processing?

Reply:
The bigger WTF would be config versoin in the db that's older than current :thinking:. So maybe add `>=` here instead of `==`

Reply:
Actually I was afraid of situation when reload task is waiting in queue and plugin_config_version in database is somehow equal to the current version. This situation should never happen, but can cause infinite loop. `logging.error` should be enough.

Reply:
Version is a string, so for `>=` check we need to convert it to int and replace `initial` with 0 🤔